modules/Hoot/managers/api.js

```diff
@@ -1042,6 +1047,190 @@ export default class API {
+     * @param uri {String} The URI used to alert JOSM via remote control.
+     * @param goodMessage {String} Message for a successful connection.
+     * @param badMessage {String} Message for an unsuccessful connection.
+     */
+    callJosmRemoteControl(uri, goodMessage, badMessage) {
+        // Safari won't send AJAX commands to the default (insecure) JOSM port when
+        // on a secure site, and the secure JOSM port uses a self-signed certificate
+        // that requires the user to jump through a bunch of hoops to trust before
+        // communication can proceed. So for Safari only, fall back to sending JOSM
+        // requests via the opening of a separate window instead of AJAX.
+        // Source: https://github.com/osmlab/maproulette3
+        let safariWindowReference = null;
+        if (window.safari) {
+            return new Promise((resolve) => {
+                if (safariWindowReference && !safariWindowReference.closed) {
+                    safariWindowReference.close();
+                }
+
+                safariWindowReference = window.open(uri);
+
+                // Close the window after 1 second and resolve the promise
+                setTimeout(() => {
+                    if (safariWindowReference && !safariWindowReference.closed) {
+                        safariWindowReference.close();
+                    }
+                    resolve(true);
+                }, 1000);
+            });
+        }
+        return fetch(uri)
+            .then(response => {
+                let message = goodMessage;
+                Hoot.message.alert({
+                    message: message,
+                    type: 'success'
+                });
+                return response.status === 200;
+            })
+            .catch(() => {
+                let message = badMessage;
+                Hoot.message.alert({
+                    message: message,
+                    type: 'error'
+                });
+                return false;
+            });
+    }
+
+    getOutputName(type, input) {
+        let output;
+        switch (type) {
+            case 'Datasets': {
+                output = 'hoot_export_datasets';
+                break;
+            }
+            default: {
+                output = input;
+                break;
+            }
+        }
+        return output;
+    }
+
+    getInputType(type) {
+        switch (type) {
+            case 'Dataset': {
+                type = 'db';
+                break;
+            }
+            case 'Datasets': {
+                type = 'dbs';
+                break;
+            }
+            default: break;
+        }
+        return type;
+    }
+
+    getTranslationPath(translations) {
+        let selectedTranslation = 'OSM';
+        const translation = translations.find(t => t.name === selectedTranslation);
+        return !translation.hasOwnProperty('path') ? translation.exportPath : translation.path;
+    }
```

Comment:
these aren't really api methods, so would be better scoped as functions under `openDataInJosm`